wp-content/themes/oceanwp-child/functions.php

```diff
@@ -252,11 +243,55 @@ function add_related_news_to_initiative_pages() {
 			<?php } ?>
 
 			</div>
-        <?php
-             
+        <?php            
          
-        }
-         
-    }
- 
+        }         
+    } 
+}
+
+function isUserLoggedIn() {
+	if ( !is_user_logged_in() ) {
+
+		wp_redirect('http://localhost/ggrc_website/'); 
+		
+		exit;
+	}
+}
+
+add_action( 'wp_ajax_follow_post', 'follow_initiative' );
+function follow_initiative() {
+	global $wpdb;
+
+	isUserLoggedIn();
+
+	$current_user_id = get_current_user_id();
+	$postid = $_POST['postid'];
+
+	$table_name = $wpdb->prefix . 'follow_posts';     
+	$wpdb->insert($table_name, array('userID' => $current_user_id, 'postID' => $postid)); 		
+	
+}
+
+add_action( 'wp_ajax_unfollow_post', 'unfollow_initiative' );
+function unfollow_initiative() {
+	global $wpdb;     
+        
+	isUserLoggedIn();
+
+	$current_user_id = get_current_user_id();
+	$postid = $_POST['postid'];			
+	$table_name = $wpdb->prefix . 'follow_posts';     
+		
+	$wpdb->query($wpdb->prepare("UPDATE $table_name SET isFollowing = '0' WHERE userID = '$current_user_id' and postID = '$postid'"));		
+		
+}
+
+function checkIsFollowing() {
+	global $wpdb;
+
+	$current_user_id = get_current_user_id();
+	$postid= get_the_ID();
+	$posts = $wpdb->get_results("SELECT DISTINCT userID, postID FROM ggrc_follow_posts WHERE `userID` = '$current_user_id' and `postID` = '$postid' and `isFollowing` = 1");
+
+	return count($posts);
```

Comment:
If this function checks if user is following, why do we count?
Would this work the same and be clearer?
```suggestion
	return isEmpty($posts);
```
The you can just do this in the template:
`<?php if (isCurrentUserFollowing()) { ?>`

Reply:
isEmpty function doesn't seem to work so I'm checking like this -> `if (!empty(isCurrentUserFollowing()))`

Reply:
Yes, you are right. `isEmpty` is from another language :)
In PHP it is `empty`. Updated my suggestion